tts-service/service/service_test.go

```diff
@@ -189,6 +189,85 @@ func TestService(t *testing.T) {
 			So(actions[0], ShouldEqual, "persistence.create")
 			So(actions[1], ShouldEqual, "persistence.get")
 		})
+
+		Convey("Delete by Id should return an error if media file not exists", func() {
+			actions := []string{}
+
+			//given
+			mock := mock("abc", ttsData{"Hello,World", "EN", StatusPending.String(), "media"})
+			s := New(mock, mock)
+
+			//when
+			err := s.Delete("abc")
+
+			//then
+			So(err, ShouldNotBeNil)
+			notFound, ok := err.(ObjectNotFoundError)
+
+			So(ok, ShouldBeTrue)
+			So(notFound.Message, ShouldEqual, "TTS with ID: 'media' doesn't exist")
+
+			//Ensure all operations in the backgrounds completed...
+			actions = readBlocking(actions, mock.recordChan)
+			actions = readBlocking(actions, mock.recordChan)
+
+			//Verify interaction
+			So(actions[0], ShouldEqual, "persistence.get")
+			So(actions[1], ShouldEqual, "tts.Engine.Delete")
+
+		})
+
+		Convey("Delete by Id should return an error if media metadata not exists", func() {
```

Comment:
Again - remember that even if one of the steps fails, the code should try BOTH operations. Of course, API should return a single error which brings a question: what to return if both operations fail? An error with "details" array would be good for that case.